The job **`x86_64-gnu-miri`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/24286607254/job/70916971706) [(plain enhanced)](https://triage.rust-lang.org/gha-logs/rust-lang/rust/70916971706) [(plain)](https://github.com/rust-lang/rust/commit/3fc647eafca50e1869963620fc16dcc8c6920f63/checks/70916971706/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
   Compiling constant_time_eq v0.3.1
error[E0599]: no method named `push_mut` found for struct `RefCell<T>` in the current scope
   --> compiler/rustc_arena/src/lib.rs:290:37
    |
290 |             let chunk = self.chunks.push_mut(ArenaChunk::<T>::new(new_cap));
    |                                     ^^^^^^^^ method not found in `RefCell<Vec<ArenaChunk<T>>>`
    |
note: the method `push_mut` exists on the type `Vec<ArenaChunk<T>>`
   --> /rustc/ad726b5063362ec9897ef3d67452fc5606ee70fa/library/alloc/src/vec/mod.rs:1023:4
help: use `.borrow_mut()` to mutably borrow the `Vec<ArenaChunk<T>>`, panicking if any borrows exist
    |
290 |             let chunk = self.chunks.borrow_mut().push_mut(ArenaChunk::<T>::new(new_cap));
    |                                    +++++++++++++

For more information about this error, try `rustc --explain E0599`.
[RUSTC-TIMING] rustc_arena test:false 0.075
error: could not compile `rustc_arena` (lib) due to 1 previous error
```

</details>

--- a/compiler/rustc_arena/src/lib.rs
+++ b/compiler/rustc_arena/src/lib.rs
@@ -287,10 +287,9 @@ impl<T> TypedArena<T> {
             // Also ensure that this chunk can fit `additional`.
             new_cap = cmp::max(additional, new_cap);
 
-            let mut chunk = ArenaChunk::<T>::new(new_cap);
+            let chunk = chunks.push_mut(ArenaChunk::<T>::new(new_cap));
             self.ptr.set(chunk.start());
             self.end.set(chunk.end());
-            chunks.push(chunk);
         }
     }
 
@@ -419,7 +418,7 @@ impl DroplessArena {
             // Also ensure that this chunk can fit `additional`.
             new_cap = cmp::max(additional, new_cap);
 
-            let mut chunk = ArenaChunk::new(align_up(new_cap, PAGE));
+            let chunk = chunks.push_mut(ArenaChunk::new(align_up(new_cap, PAGE)));
             self.start.set(chunk.start());
 
             // Align the end to DROPLESS_ALIGNMENT.
@@ -430,8 +429,6 @@ impl DroplessArena {
             debug_assert!(chunk.start().addr() <= end);
 
             self.end.set(chunk.end().with_addr(end));
-
-            chunks.push(chunk);
         }
     }
 

--- a/library/proc_macro/src/bridge/arena.rs
+++ b/library/proc_macro/src/bridge/arena.rs
@@ -58,11 +58,10 @@ impl Arena {
         // Also ensure that this chunk can fit `additional`.
         new_cap = cmp::max(additional, new_cap);
 
-        let mut chunk = Box::new_uninit_slice(new_cap);
+        let chunk = chunks.push_mut(Box::new_uninit_slice(new_cap));
         let Range { start, end } = chunk.as_mut_ptr_range();
         self.start.set(start);
         self.end.set(end);
-        chunks.push(chunk);
     }
 
     /// Allocates a byte slice with specified size from the current memory

